core/src/test/java/org/apache/calcite/util/SourceTest.java

```diff
@@ -67,12 +70,27 @@ private static String getRootPrefix() {
     }
   }
 
+  @Test public void testFileToUrl() throws URISyntaxException {
+    final String path = "abc.txt";
+    Source fileSource = file(null, path);
+    URL url = fileSource.url();
+
+    assertFalse(url == null);
```

Comment:
Please don't do that. It would be extremely hard to understand the nature of the failure if that assert fails in CI.

Please use the appropriate methods, and add messages so the failure is easy to understand by looking at CI output.

Reply:
Thanks for the good suggestion. I have revised the code to include an error message.